core-bundle/src/Security/Authentication/ContaoLoginAuthenticationListener.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Security\Authentication;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Http\Firewall\UsernamePasswordFormAuthenticationListener;
+
+class ContaoLoginAuthenticationListener extends UsernamePasswordFormAuthenticationListener
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function requiresAuthentication(Request $request)
+    {
+        if (!$request->isMethod('POST') || 0 !== strpos((string) $request->request->get('FORM_SUBMIT'), 'tl_login')) {
```

Comment:
Why are you using `strpos()` here instead of a simple `===`?

Reply:
the frontend login module uses `tl_login_%ID%` as the form submit value.

Reply:
Oh yes, I recall. 👍 

Personally, I would prefer to write the method like this:

```php
protected function requiresAuthentication(Request $request): bool
{
    if (!$request->isMethod('POST') || !$request->request->has('FORM_SUBMIT')) {
        return false;
    }

    return 0 === strpos(request->request->get('FORM_SUBMIT'), 'tl_login');
}
```

WDYT?

Reply:
Oh, and do we have to account for `tl_loginfoo` as well? The method would return `true` currently. Probably change it to `preg_match('/^tl_login(_\d+)?$/', $request->request->get('FORM_SUBMIT'))`?

Reply:
There's no real case of `tl_loginfoo` anywhere, so I don't think we need that complication.